AT-9931 EDA Script, getCLINData method refactored (do not fail fast if CLIN is missing PoP dates)

stephenhayes-ccpo opened the pull request:
optional chaining & nullish coalescing to avoid throwing if CLIN is missing PoP dates
looping reduced

stephenhayes-ccpo committed:
AT-9931 
EDA Script, getCLINData method refactored (do not fail fast if CLIN is missing PoP dates)
looping reduced
optional chaining & nullish coalescing

stephenhayes-ccpo committed:
AT-9931 linting/comments

stephenhayes-ccpo committed:
AT-9931
removed truthy checks for CLIN dates / funds

jeffsegal-ccpo commented on f600233d1b154d507b782f84604bcb12/update/sys_script_include_abc17e0edba4e918b1227ea5f39619fc.xml:
```diff
@@ -496,31 +496,33 @@ Eda.prototype = {
+                    totalAmount
+                };
+                // Gather Obligated Amount from "INFOSLIN" entries
+            } else if (type === "INFOSYS" && lineItem?.LineItemAmounts) {
```
Shouldn't this be `type === "INFOSLIN"`?

jeffsegal-ccpo commented on f600233d1b154d507b782f84604bcb12/update/sys_script_include_abc17e0edba4e918b1227ea5f39619fc.xml:
```diff
@@ -40,6 +40,7 @@ Eda.prototype = {
             "3000000000000": "GCP",
             "4000000000000": "Oracle",
             "5000000000000": "Non-JWCC",
+			"6000000000000": "Busted"
```
Is this referring to a file that you expect to exist in the Dummy Data Table? If so, I don't see it in this PR.

stephenhayes-ccpo committed:
AT-9931
collect errors on missing CLIN data; if any CLIN is missing data, return 400
linting

jeffsegal-ccpo reviewed (approved):
Looks great now.
